pkg/webhook/v1beta1/webhook.go

```diff
@@ -32,10 +32,14 @@ import (
 	"github.com/kubeflow/katib/pkg/webhook/v1beta1/pod"
 )
 
-func AddToManager(m manager.Manager, port int32, serviceName string) error {
+func AddToManager(m manager.Manager, port int32, webhookInstaller bool, serviceName string) error {
 	so := webhook.ServerOptions{
```

Comment:
In case of custom WebHook configuration, should Katib controller create the webhook server ?

Maybe in case of custom webhooks we should disable adding to the manager here: 
https://github.com/kubeflow/katib/blob/master/cmd/katib-controller/v1beta1/main.go#L117-L121?

WDYT @gaocegege @johnugeorge ?

Reply:
The scenario that I've got is that I'd like katib-controller to respond to webhook requests, but I'd like to set up the webhooks and Service resources myself, instead of katib-controller creating them. AFAICT, if we disabled adding to the manager as above, katib-controller wouldn't be listening to webhook requests. Is that correct?

Reply:
@knkski We can follow the [KFServing way](https://github.com/kubeflow/kfserving/blob/master/cmd/manager/main.go#L156-L178), but I believe we should update the controller-runtime pkg version for that.

Did you check my comment above: https://github.com/kubeflow/katib/pull/1424#pullrequestreview-579237771 ?
Maybe instead of adding another flag to the controller we should start with YAMLs for the webhooks?

/cc @gaocegege @johnugeorge

Reply:
@andreyvelich: Is there an example somewhere else for using YAML files for the webhooks that you had in mind?

Reply:
@knkski For example, KFServing: https://github.com/kubeflow/kfserving/blob/master/config/webhook/manifests.yaml, Tekton Pipeline: https://github.com/tektoncd/pipeline/blob/master/config/500-webhooks.yaml.

Reply:
If we need to update controller-runtime's version maybe we need to update k8s version first. tf-operator now depends on k8s 1.16, we are trying to remove direct deps to k8s. Ref https://github.com/kubeflow/tf-operator/pull/1235

Reply:
@gaocegege We currently use Kubernetes API from 1.12: https://github.com/kubeflow/katib/blob/master/Gopkg.toml#L57-L59.

I am fine with updating K8s version with controller-runtime, since currently Katib webhooks doesn't work with K8s 1.19, check issue here: https://github.com/kubeflow/katib/issues/1395#issuecomment-750373088.